source/material_model/nondimensional.cc

```diff
@@ -0,0 +1,222 @@
+                             Patterns::Double (0),
+                             "The value of the specific heat $cp$. "
+                             "Units: $J/kg/K$.");
+          prm.declare_entry ("Viscosity temperature prefactor", "0.0",
+                             Patterns::Double (0),
+                             "Exponential temperature prefactor for viscosity.");
+          prm.declare_entry ("Viscosity depth prefactor", "0.0",
+                             Patterns::Double (0),
+                             "Exponential depth prefactor for viscosity.");
+          prm.declare_entry ("Use TALA", "false",
+                             Patterns::Bool (),
+                             "Whether to use the TALA instead of the ALA "
+                             "approximation.");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    Nondimensional<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Nondimensional model");
+        {
+          reference_rho   = prm.get_double ("Reference density");
+          exponential_viscosity_temperature_prefactor = prm.get_double ("Viscosity temperature prefactor");
+          exponential_viscosity_depth_prefactor = prm.get_double ("Viscosity depth prefactor");
+          Di              = prm.get_double ("Di");
+          Ra              = prm.get_double ("Ra");
+          gamma           = prm.get_double ("gamma");
+          tala            = prm.get_bool ("Use TALA");
+          reference_specific_heat = prm.get_double ("Reference specific heat");
+
+          compressible = (Di!=0.0);
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::temperature;
+      this->model_dependence.density = NonlinearDependence::pressure | NonlinearDependence::temperature;
+      this->model_dependence.compressibility = NonlinearDependence::none;
+      this->model_dependence.specific_heat = NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = NonlinearDependence::none;
+
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(Nondimensional,
+                                   "nondimensional",
+                                   "A material model for nondimensionalized "
+                                   "computations for compressible or incompressible "
+                                   "computations defined through Rayleigh number Ra "
+                                   "and Dissipation number Di. This model is made "
+                                   "to be used with the Boussinesq, ALA, or TALA "
+                                   "formulation.")
```

Comment:
can you add formulas that describe how the density and other computed quantities look like? I think that would be useful to understand what the "viscocity temperature prefactor" is. You could then reference the various parameters to symbols that appear in the material model formulas.